orangecontrib/bioinformatics/annotation/annotate_samples.py

```diff
@@ -0,0 +1,202 @@
+from collections import defaultdict
+
+import numpy as np
+from Orange.data import Domain, ContinuousVariable, Table
+
+from orangecontrib.bioinformatics.ncbi.gene import GeneInfo
+from orangecontrib.bioinformatics.utils import statistics
+from scipy.stats.mstats import rankdata
+from scipy.stats import hypergeom
+
+from orangecontrib.bioinformatics.widgets.utils.data import TAX_ID
+
+
+class AnnotateSamples:
+    """
+    AnnotateSamples is class used for the annotation of data items with the
+    labels Mann-Whitney U test for selecting important values and
+    the Hyper-geometric for assigning the labels.
+
+    Example of use:
+
+    >>> from Orange.data import Table
+    >>> from orangecontrib.bioinformatics.utils import serverfiles
+    >>> from orangecontrib.bioinformatics.annotation.annotate_samples import \
+    ...     AnnotateSamples
+    >>> from orangecontrib.bioinformatics.widgets.utils.data import TAX_ID
+    >>>
+    >>> data = Table("https://datasets.orange.biolab.si/sc/aml-1k.tab.gz")
+    >>> data.attributes[TAX_ID] = "9606"  # table needs to have an organism ID
+    >>> markers_path = serverfiles.localpath_download(
+    ...     'marker_genes','panglao_gene_markers.tab')
+    >>> marker = Table(markers_path)
+    >>>
+    >>> # filter only human markers
+    >>> from Orange.data.filter import FilterString, Values
+    >>> f = FilterString("Organism", FilterString.Equal, "Human")
+    >>> markers = Values([f])(markers)
+    >>>
+    >>> annotator = AnnotateSamples(p_value_th=0.05)
+    >>> annotations = annotator.annotate_samples(data, markers)
+
+    Attributes
+    ----------
+    p_value_th : float
+        A threshold for accepting the annotations. Annotations that has FDR
+        value bellow this threshold are used.
+    p_value_fun : callable, optional (defaults: statistics.Binomial().p_value)
+        A function that calculates p-value. It can be either
+        statistics.Binomial().p_value or hypergeom.sf.
+    """
+
+    def __init__(self, p_value_th, p_value_fun=statistics.Binomial().p_value):
```

Comment:
For `p_value_fun`, I think it is better to pass a string constant (named e.g. TEST_HYPERGEOMETRIC, TEST_BINOMIAL) rather than a function. This way, there is no need to import a module when using this class in scripts.